library/src/main/java/com/novoda/downloadmanager/VersionOneToVersionTwoMigrator.java

```diff
@@ -81,28 +75,14 @@ public void migrate() {
 
         Log.d(TAG, "all data migrations are COMMITTED, about to delete the old database, time is " + System.nanoTime());
 
-        internalMigrationStatus.markAsDeleting();
-        notifyOfProgress(internalMigrationStatus);
+        migrationStatus.markAsDeleting();
+        migratorUpdateCallback.onUpdate(migrationStatus);
         Log.d(TAG, "all traces of v1 are ERASED, time is " + System.nanoTime());
         database.close();
 
         database.deleteDatabase();
-        internalMigrationStatus.markAsComplete();
-        notifyOfProgress(internalMigrationStatus);
-    }
-
-    private void notifyOfProgress(MigrationStatus migrationStatus) {
```

Comment:
I thought this just added noise to this class and the `Service` or some other object can be responsible for deciding how to display a notification based on a status.